robottelo/ui/trend.py

```diff
@@ -0,0 +1,90 @@
+# -*- encoding: utf-8 -*-
+"""
+Implements Monitor->Trends UI
+"""
+
+from robottelo.ui.base import Base, UIError, UINoSuchElementError
+from robottelo.ui.locators import locators, common_locators
+from robottelo.ui.navigator import Navigator
+from selenium.webdriver.support.select import Select
+
+
+class Trend(Base):
+    """Provides the CRUD functionality for Trends."""
+
+    def create(self, trend_type, trendable, name):
+        """Creates the trend."""
+        self.wait_until_element(locators["trend.new"]).click()
+
+        if self.wait_until_element(locators["trend.type"]):
+            Select(
+                self.find_element(locators["trend.type"])
+            ).select_by_visible_text(trend_type)
+
+            if trendable is not None:
+                Select(
+                    self.find_element(locators["trend.trendable"])
+                ).select_by_visible_text(trendable)
+
+            if name is not None and trendable is not None:
+                self.find_element(locators["trend.name"]).send_keys(name)
+
+        self.find_element(common_locators["submit"]).click()
+        self.wait_for_ajax()
+
+    def update(self, trend_name, entity_name, entity_value):
+        """Updates a trend."""
+        element = self.search(trend_name)
+        if element:
+            strategy = locators["trend.edit"][0]
+            value = locators["trend.edit"][1]
+            edit_fld = self.wait_until_element((strategy, value % trend_name))
+            edit_fld.click()
+            strategy = locators["trend.edit_entity"][0]
+            value = locators["trend.edit_entity"][1]
+            txt_fld = self.wait_until_element((strategy, value % entity_name))
+            if txt_fld:
+                txt_fld.clear()
+                txt_fld.send_keys(entity_value)
+            elif txt_fld is None:
+                raise UINoSuchElementError(
+                    u'Could not find the entity "{0}" for update procedure.'
+                    .format(entity_name)
+                )
+            self.find_element(common_locators["submit"]).click()
+            self.wait_for_ajax()
```

Comment:
This can be rewritten a little bit if you check for error conditions instead of checking for success conditions:

```python
        element = self.search(trend_name)
        if element is None:
            return  # FIXME: raise an exception instead?
        strategy = locators['trend.edit'][0]
        value = locators['trend.edit'][1]
        edit_fld = self.wait_until_element((strategy, value % trend_name))
        edit_fld.click()
        strategy = locators['trend.edit_entity'][0]
        value = locators['trend.edit_entity'][1]
        txt_fld = self.wait_until_element((strategy, value % entity_name))
        if txt_fld is None:
            raise UINoSuchElementError(
                u'Could not find the entity "{0}" for update procedure.'
                .format(entity_name)
            )
        txt_fld.clear()
        txt_fld.send_keys(entity_value)
        self.find_element(common_locators['submit']).click()
        self.wait_for_ajax()
```